src/streamlink/plugins/itvplayer.py

```diff
@@ -1,213 +1,83 @@
+import json
+import logging
 import re
-from uuid import uuid4
-
-try:
-    import xml.etree.cElementTree as ET
-except ImportError:
-    import xml.etree.ElementTree as ET
 
+from streamlink.compat import urljoin
 from streamlink.plugin import Plugin
-from streamlink.plugin.api import http
-from streamlink.stream import RTMPStream, HDSStream
-from streamlink.compat import urlparse, unquote
+from streamlink.plugin.api import http, useragents, validate
+from streamlink.plugin.api.utils import itertags
+from streamlink.stream import HLSStream
 
-ITV_PLAYER_USER_AGENT = 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/32.0.1700.107 Safari/537.36'
-LIVE_SWF_URL = "http://www.itv.com/mediaplayer/ITVMediaPlayer.swf"
-ONDEMAND_SWF_URL = "http://www.itv.com/mercury/Mercury_VideoPlayer.swf"
-CHANNEL_MAP = {'itv': 1, 'itv2': 2, 'itv3': 3, 'itv4': 4, 'itvbe': 8, 'citv': 7}
-_url_re = re.compile(r"http(?:s)?://(?:www.)?itv.com/hub/(?P<stream>.+)")
+log = logging.getLogger(__name__)
 
 
 class ITVPlayer(Plugin):
-    def __init__(self, url):
-        Plugin.__init__(self, url)
-        match = _url_re.match(url)
-        self._stream = match and match.groupdict()["stream"]
+    _url_re = re.compile(r"https?://(?:www.)?itv.com/hub/(?P<stream>.+)")
+    _video_info_schema = validate.Schema({
+        "StatusCode": 200,
+        "AdditionalInfo": {
+            "Message": validate.any(None, validate.text)
+        },
+        "Playlist": {
+            "VideoType": validate.text,
+            "Video": {
+                "Subtitles": validate.any(None, [{
+                    "Href": validate.url(),
+                }]),
+                "Base": validate.url(),
+                "MediaFiles": [
+                    {"Href": validate.text,
+                     "KeyServiceUrl": validate.any(None, validate.url())}
+                ]
+            }
+        }
+    })
 
     @classmethod
-    def can_handle_url(self, url):
-        match = _url_re.match(url)
-        return match
-
-    @property
-    def channel_id(self):
-        if self._stream in CHANNEL_MAP:
-            return CHANNEL_MAP[self._stream]
+    def can_handle_url(cls, url):
+        match = cls._url_re.match(url)
+        return match is not None
 
     @property
-    def production_id(self):
-        if self._stream not in CHANNEL_MAP:
-            res = http.get(self.url, verify=False)
-            production_id_match = re.findall(r'data-video-production-id="(.+?)"', res.text)
-            if production_id_match:
-                return unquote(production_id_match[0])
-            else:
-                self.logger.error(u"No production ID found, has the page layout changed?")
+    def device_info(self):
+        return {"user": {},
+                "device": {"manufacturer": "Chrome", "model": "66",
```

Comment:
it's chrome here

in the **long term it might break** because of two different User-Agents

`useragents.CHROME` is probably outdated
maybe update it or create a `CHROME_66`

Reply:
We should update them all :) I’ll make it consistent too.

Reply:
It doesn't look like they actually check the device info, it works fine if you remove it.